multiraft/multiraft.go

```diff
@@ -654,10 +654,21 @@ func (s *state) createGroup(groupID uint64) error {
 
 func (s *state) removeGroup(op *removeGroupOp) {
 	// Group creation is lazy and idempotent; so is removal.
-	if _, ok := s.groups[op.groupID]; !ok {
+	log.V(3).Infof("removing group %d", op.groupID)
+	g, ok := s.groups[op.groupID]
+	if !ok {
 		op.ch <- nil
 		return
 	}
+
+	// Cancel commands which are still in transit.
+	for cmdID, prop := range g.pending {
+		if prop.ch != nil {
+			prop.ch <- util.Errorf("group deleted")
```

Comment:
This needs to return a retryable error to the client. To maintain separation of multiraft, create a new error type here to return, and then have the client convert multiraft.GroupDeletedError to a new proto.RangeMovedError (which should be Retryable). This way the client will retry and find the new home of this range.